country-dashboard/src/App.tsx

```diff
@@ -1,28 +1,253 @@
+  const [lang, setLang] = useState<any[]>([]);
+
+  // state to store region array
+  const [region, setRegion] = useState<any[]>([]);
+
+  // state for the search field to filter
+  const [search, setSearch] = useState("");
+  // state for the currency field to filter
+  const [currency, setCurrency] = useState("Any Currency");
+  // state for the search field to filter
+  const [language, setLanguage] = useState("Any Language");
+  // state for the search field to filter
+  const [reg, setregion] = useState("Any Region");
+
+  // function to get  the countires , currencies , languages , region
+  useEffect(() => {
+    (async () => {
+      const response = await fetch("https://restcountries.com/v3.1/all");
+      const data = await response.json();
+      setCountries(data);
+      setCurr(
+        data
+          ? Array.from(
+              new Set(
+                data
+                  .map((item: any) => {
+                    // Check if currencies key exists and is not empty
+                    if (
+                      item.currencies &&
+                      Object.keys(item.currencies).length > 0
+                    ) {
+                      return Object.keys(item.currencies)[0];
+                    } else {
+                      return null; // or any other value you want to use for empty currencies
+                    }
+                  })
+                  .filter((curr: null) => curr !== null)
+              )
+            )
+          : []
+      );
+
+      setLang(
+        data
+          ? Array.from(
+              new Set(
+                data
+                  .map((item: any) => {
+                    if (
+                      item.languages &&
+                      Object.keys(item.languages).length > 0
+                    ) {
+                      return Object.entries(item.languages)[0][1];
+                    } else {
+                      return null;
+                    }
+                  })
+                  .filter(
+                    (lang: string | any[] | null) =>
+                      lang !== null && lang.length > 0
+                  )
+              )
+            )
+          : []
+      );
+
+      setRegion(
+        data
+          ? Array.from(
+              new Set(
+                data
+                  .map((item: any) => item.region)
+                  .filter(
+                    (region: string | any[]) => region && region.length > 0
+                  )
+              )
+            )
+          : []
+      );
+    })();
+  }, []);
+
+  function handleSearch(event: any) {
+    const countryName = event.target.value;
+    setSearch(countryName);
+
+    // Check if the input contains only alphabetic characters or is empty
+    if (!/^[a-zA-Z]*$/.test(countryName.trim())) {
+      // If input contains numbers or special characters, do not perform the search
+      return;
+    }
+
+    // If search field is empty, fetch all countries
```

Comment:
While your filtering works, it does not handle AND conditions. For example, if I enter USD currency and language French, it returns countries that speak french but do not use USD. This doesn't work. 

What should be done is that when you get the response for countries by some filter, you should combine the result with the existing data (which may already be filtered) and only include countries that are both in the current data list (already filtered) and the countries in the search response. 

-10

Reply:
Shrinjay for filtering all the selected values the question was asked by the NANCY in the call , and you said do it only single filter . The combine filter will be addressed in furure. Thats why ....combine filter AND not working

Reply:
Ohh right, I'm really sorry, I just remembered this discussion. I apologize, will give back points.